Blog post: 2024 in review (#5775)
open-telemetry/opentelemetry.io: content/en/blog/2024/year-in-review.md

---
title: Opentelemetry.io 2024 review
linkTitle: Year in review

> date: 2024-12-18

Reviewer: Oops, I forgot that we don't preview future posts anymore. Setting date back:

```suggestion
date: 2024-12-12 # Date to be set when we actually publish
```

> author:
>   >- # If you have only one author, then add the single name on this line in quotes.
>   [Severin Neumann](https://github.com/svrnm) (Cisco)

Author: ```suggestion
author:
  [Severin Neumann](https://github.com/svrnm) (Cisco),
  [Patrice Chalin](https://github.com/chalin/) (CNCF),
  [Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)
```

> date: 2024-12-12 # Put the current date, we will keep the date updated until your PR is merged
> author:
>   [Severin Neumann](https://github.com/svrnm) (Cisco),
>   [Patrice Chalin](https://github.com/chalin/) (CNCF),
>   [Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)
> sig: SIG Comms
> cSpell:ignore: opentelemetrybot

Reviewer: ```suggestion
date: 2024-12-18
author: >-
  [Severin Neumann](https://github.com/svrnm) (Cisco),
  [Patrice Chalin](https://github.com/chalin/) (CNCF),
  [Tiffany Hrabusa](https://github.com/tiffany76) (Grafana Labs)
sig: Comms
cSpell:ignore: Chalin Hrabusa opentelemetrybot
```

For the next year, we plan to rework the way we introduce OpenTelemetry to
beginners. If you are interested in helping, [join us here](https://github.com/open-telemetry/community/pull/2427/).

## Intriguing insights

Here are some of the most interesting statistics about the OpenTelemetry website:

- The most changed content file is `content/en/docs/collector/_index.md` with 91
  updates since the file was created.
- With 511 commits, 26,765 additions and 9,734 deletions the
  [opentelemetrybot](https://github.com/opentelemetrybot) is our top-4
  contributor.
- The word OpenTelemetry occurs 7313 times in the source files of the English
  website. With that it is the 3rd most frequent word right after 'the' and
  'to'. The world 'collector' is used 3186 times and at the 11th place!
- The PR with the most comments this year and also for all time is
  [Blog post for OpenTelemetry Generative AI updates](https://github.com/open-telemetry/opentelemetry.io/pull/5575),
  with 150 comments. Close 2nd place goes to
  [[pt] Translate /pt/docs/languages/go/instrumentation](https://github.com/open-telemetry/opentelemetry.io/pull/5380)
  with 146 comments.

> -

Author: ```suggestion
```

---

> title: '2024: The Year in Review for opentelemetry.io'
> linkTitle: 'Year in Review'

Reviewer: ```suggestion
title: Opentelemetry.io 2024 review
linkTitle: Year in review
```

date: 2024-12-12 # Put the current date, we will keep the date updated until your PR is merged
author:
  >- # If you have only one author, then add the single name on this line in quotes.

>   [Severin Neumann](https://github.com/svrnm) (Cisco),

Reviewer: ```suggestion
  [Severin Neumann](https://github.com/svrnm) (Cisco)
```

sig: SIG Comms

> ---

Reviewer: ```suggestion
cSpell:ignore: opentelemetrybot
---
```

> As 2024 comes to a close, we'd like to take the time to take a look back at it,
> and share some insights and accomplishments of SIG Communcation, which is
> responsible for running this website, blog and documentation.

Reviewer: ```suggestion
As 2024 draws to a close, we’d like to reflect on the year and share some
insights and accomplishments from SIG Communications, the team responsible for
managing this website, blog, and documentation.
```

## Contributions

> In
> [December 2022](https://github.com/open-telemetry/opentelemetry.io/releases/tag/2022.12)
> we started publishing monthly releases of our website on GitHub, to have a
> regular summary of contributions. Based on this data we can do a long time
> comparison on contributions, and if we look at the time between
> [December, 2022 and November, 2023](https://github.com/open-telemetry/opentelemetry.io/compare/2022.12...2023.11)
> and compare it with
> [December, 2023 to November, 2024](https://github.com/open-telemetry/opentelemetry.io/compare/2023.12...2024.11),
> we see an upwards trend in commits from 1,011 to 1,340, and contributors from 92
> to 106. The only metric which has gone down is the number of files changed,
> which was 1,864 previously and went down to 1,624.
> 
> Overall there have been 768 contributors and 3,982 commits across 3,824 merged
> pull requests since the repository was created in April, 2019.
> 
> We thank every contributor for helping to build and improve the OpenTelemetry
> website!

Reviewer: ```suggestion
In
[December 2022](https://github.com/open-telemetry/opentelemetry.io/releases/tag/2022.12),
we began publishing monthly releases of our website on GitHub to provide regular
summaries of contributions. Using this data, we can perform long-term
comparisons of contributions. For example, when comparing the period from
[December 2022 to November 2023](https://github.com/open-telemetry/opentelemetry.io/compare/2022.12...2023.11)
with
[December 2023 to November 2024](https://github.com/open-telemetry/opentelemetry.io/compare/2023.12...2024.11),
we see an upward trend: commits increased from 1,011 to 1,340, and contributors
grew from 92 to 106. The only metric that decreased was the number of files
changed, which dropped from 1,864 to 1,624.

Since the repository's creation in April 2019, there have been 768 contributors
and 3,982 commits across 3,824 merged pull requests.

We extend our heartfelt thanks to every contributor for helping build and
improve the OpenTelemetry website!
```

## User analytics

Based on
[data collected with Google Analytics](https://lookerstudio.google.com/s/tSTKxK1ECeU)
the [opentelemetry.io](/) website had over 12 million views across 4 million

> sessions this year. The year before it was visited with almost 10 million viewes

Reviewer: ```suggestion
sessions this year. The year before it was visited with almost 10 million views
```

across over 3 million sessions, which means there have been ~16% more views
year-over-year.

With almost 2,9 million views the [landing page](/) is the most popular page,
followed by the [Collector page](/docs/collector) with over 400,000 views.

## Localizations

> A major accomplishment this year was, that
> [we went multilingual](/blog/2024/docs-localized/). Localization teams are
> translating pages from English to [Chinese](/zh), [French](/fr),
> [Japanese](/ja), [Portuguese](/pt) and [Spanish](/es), with a total sum of 122
> pages translated so far!
> 
> We thank everyone who has contributed translations, and we are excited to be
> able to provide these language options that improve the OpenTelemetry user
> experience.

Reviewer: ```suggestion
A major accomplishment this year was that [we went multilingual](/blog/2024/docs-localized/). Localization teams have been translating pages from English into several languages, with a total of 122 pages translated so far! The available languages include:

- [Chinese](/zh)  
- [French](/fr)  
- [Japanese](/ja)  
- [Portuguese](/pt)  
- [Spanish](/es)  

We thank everyone who contributed to the translations and are excited to offer these language options, enhancing the OpenTelemetry user experience.
```

## Information architecture changes

> Another big change this year was a change in our information architecture: we
> renamed the documentation section `Instrumentations` to `Languages APIs & SDKs`
> to make it clearer to end-users what to expect in that section. We also moved
> out the pages around `Automatic Instrumentation` into a section called
> `Zero-code instrumentation`, to provide a clearer separation of using the APIs
> and SDKs for instrumentation, and using instrumentation tools like a Java agent,
> that add OpenTelemetry from the outside.
> 
> As a follow up to that change, the Java SIG updated the overall structure of
> their API and SDK documentation, to fit better into this new architecture.
> 
> For the next year, we plan to rework the way how we introduce OpenTelemetry to
> new starters, if you are interested in helping, you can
> [join us here](https://github.com/open-telemetry/community/pull/2427/).

Reviewer 2: Should we also say something about how we'd like to extend the structure established by the Java SIG to the documentation for other languages? 

```suggestion
We made big changes in our information architecture this year. We
renamed the documentation section `Instrumentations` to `Language APIs & SDKs`
to make it clearer to end users what to expect in that section. We also moved the `Automatic Instrumentation` content into a section called
`Zero-code instrumentation` to provide a clearer separation between API
and SDK instrumentation and instrumentation tools like a Java agent
that inject OpenTelemetry from the outside.

As a follow up to the instrumentation page changes, the Java SIG updated the overall structure of
their API and SDK documentation to fit better into this new architecture.

For the next year, we plan to rework the way we introduce OpenTelemetry to
beginners. If you are interested in helping, [join us here](https://github.com/open-telemetry/community/pull/2427/).
```

> ## Curious facts

Reviewer 2: ```suggestion
## Intriguing insights
```

There are many statistics we can create about our project, but only some of them

> are worth sharing, because they are curious:

Reviewer 2: Here are some of the most interesting statistics about the OpenTelemetry website:

Author: ```suggestion
Here are some of the most interesting statistics about the OpenTelemetry website:
```

> - The most changed content file is `content/en/docs/collector/_index.md` with 91
>   updates since the file was created.
> - With 511 commits, 26,765 additions and 9,734 deletions the
>   [opentelemetrybot](https://github.com/opentelemetrybot) is our top-4
>   contributor.
> - The word OpenTelemetry occurs 7313 times in the source files of the English
>   website. With that it is the 3rd most frequent word right after 'the' and
>   'to'. The world 'collector' is used 3186 times and at the 11th place!
> - The PR with the most comments this year and also for all time is
>   [Blog post for OpenTelemetry Generative AI updates](https://github.com/open-telemetry/opentelemetry.io/pull/5575),
>   with 150 comments. Close 2nd place goes to
>   [[pt] Translate /pt/docs/languages/go/instrumentation](https://github.com/open-telemetry/opentelemetry.io/pull/5380)
>   with 146 comments.

Reviewer 2: I wasn't sure what "top-4" meant, so I revised it. But if the term has a specific meaning that I'm unaware of, please revert!
```suggestion
- The most changed content file is `content/en/docs/collector/_index.md`, with 91
  updates since the file was created.
- With 511 commits, 26,765 additions, and 9,734 deletions the
  [opentelemetrybot](https://github.com/opentelemetrybot) is our top
  contributor.
- The word 'OpenTelemetry' occurs 7313 times in the source files of the English
  website, which makes it the 3rd most frequent word after 'the' and
  'to'. The world 'collector' is used 3186 times, putting it in 11th place!
- The PR with the most comments this year and also for all time is
  [Blog post for OpenTelemetry Generative AI updates](https://github.com/open-telemetry/opentelemetry.io/pull/5575),
  with 150 comments. Close 2nd place goes to
  [[pt] Translate /pt/docs/languages/go/instrumentation](https://github.com/open-telemetry/opentelemetry.io/pull/5380)
  with 146 comments.
```

Author: ```
- With 511 commits, 26,765 additions and 9,734 deletions the
  [opentelemetrybot](https://github.com/opentelemetrybot) is our top-4
  contributor.
```

What I want to say is that among all the contributors the bot is the one with the fourth most (?) contributions, not sure how to say that best in English

Author: ```suggestion
- The most changed content file is `content/en/docs/collector/_index.md`, with 91
  updates since the file was created.
- With 511 commits, 26,765 additions, and 9,734 deletions the
  [opentelemetrybot](https://github.com/opentelemetrybot) is the fourth most
  active contributor.
- The word 'OpenTelemetry' occurs 7313 times in the source files of the English
  website, which makes it the 3rd most frequent word after 'the' and
  'to'. The world 'collector' is used 3186 times, putting it in 11th place!
- The PR with the most comments this year and also for all time is
  [Blog post for OpenTelemetry Generative AI updates](https://github.com/open-telemetry/opentelemetry.io/pull/5575),
  with 150 comments. Close 2nd place goes to
  [[pt] Translate /pt/docs/languages/go/instrumentation](https://github.com/open-telemetry/opentelemetry.io/pull/5380)
  with 146 comments.
```

-

## People

> Having 1000+ commits per year, means that there have been equally many PRs. For
> each PR we need reviews, that ensure, that the added content fits into our
> project, is correct and written in good and plain language. We are lucky to have
> many contributors, who are taking on that responsibility: there are approvers
> and maintainers of SIGs, that co-own parts of our website, there are approvers
> for the different localizations, and there are approvers and maintainers in SIG
> Communications. A big shout out to all of them for making 2024 a successful
> year, and we are looking forward to work with you all in 2025!

Reviewer 2: ```suggestion
Having 1000+ commits per year means that there have been equally as many PRs. For
each PR, we need reviews to ensure that the added content is accurate, fits into our
project, and is well written in plain language. We are lucky to have
many contributors who take on that responsibility: approvers
and maintainers of other SIGs that co-own parts of our website, approvers
for the different localization websites, and approvers and maintainers in SIG
Communications. A big shout out to all of you for making 2024 a successful
year, and we look forward to working together in 2025!
```

> ## Call to action
> 
> If you are an OpenTelemetry end-user or contributor, or if you are just
> enthusiastic about our project, we would be excited to welcome you as a
> contributor to the website! You can help by raising issues and providing PRs! To
> get started, come by in our channel at the CNCF slack or join one of our SIG
> meetings, every other Monday at 10 PST.

Reviewer 2: ```suggestion
## Join us

If you're an end user, a contributor, or just enthusiastic about OpenTelemetry, we welcome your contributions to the website. You can contribute by raising issues, joining discussions, or making PRs. Join our [channel](https://cloud-native.slack.com/archives/C02UN96HZH6) at the [CNCF Slack](https://slack.cncf.io/) and come to our [SIG meetings](https://docs.google.com/document/d/1wW0jLldwXN8Nptq2xmgETGbGn9eWP8fitvD5njM-xZY), every other Monday at 10:00 a.m. Pacific time. With your help, we can make 2025 another successful year for [opentelemetry.io](https://opentelemetry.io/)!
```

---
title: Opentelemetry.io 2024 review
linkTitle: Year in review

> date: 2024-12-16 # FIXME: ensure this reflects the publication date

Author: ```suggestion
date: 2024-12-17
```

[December 2023 to November 2024]:
  https://github.com/open-telemetry/opentelemetry.io/compare/2023.12...2024.11
[monthly releases]: https://github.com/open-telemetry/opentelemetry.io/releases

### Which pages were the most popular?

According to our publicly available [analytics] data, [opentelemetry.io](/) was
viewed **12 million** times across 4 million sessions this year. This marks a
**16% increase** over last year's nearly 10 million views and over 3 million
sessions.

The most popular pages and sections of the documentation were:

| Page/Section             | Views | % [^1] |
| ------------------------ | ----: | -----: |
| [What is OpenTelemetry?] |  290K |   2.4% |
| [Collector]              |  1.3M |  10.5% |
| [Concepts]               |  1.2M |   9.8% |
| [Demo]                   |  829K |   6.7% |
| [Ecosystem]              |  500K |   4.0% |

[analytics]: https://lookerstudio.google.com/s/tSTKxK1ECeU
[Collector]: /docs/collector
[Concepts]: /docs/what-is-opentelemetry/
[Demo]: /docs/demo/
[Ecosystem]: /ecosystem/
[What is OpenTelemetry?]: /docs/what-is-opentelemetry/

[^1]: Percentage of the site-total 12M views.

> ### Fun Trivia <i class="fa-solid fa-lightbulb"></i> {#trivia}

Reviewer 2: ```suggestion
### Fun trivia <i class="fa-solid fa-lightbulb"></i> {#trivia}
```
